gpkit/small_scripts.py

```diff
@@ -163,41 +163,55 @@ def sort_and_simplify(exps, cs):
     return tuple(matches.keys()), cs_
 
 
-def results_table(data, title, senss=False):
-    strs = ["              | " + title]
-    for var, val in sorted(data.items(), key=lambda x: str(x[0])):
-        if isinstance(val, Iterable):
-            vector = bool(val.shape)
-            if vector:
-                if all(val == val[0]):
-                    vector = False
-                    val = val[0]
-        else:
-            vector = False
+def results_table(data, title, minval=0, printunits=True, fixedcols=True,
+                  varfmt="%4s : ", valfmt="%-8.3g", vecfmt="%-7.2g"):
+    """
+    Pretty string representation of a dict of VarKeys
+    Iterable values are handled specially (partial printing)
+
+    Parameters
+    ----------
+    data: dict whose keys are VarKey's
+        data to represent in table
+    title: string
+    minval: float
+        skip values with all(abs(value)) < minval
+    printunits: bool
+    fixedcols: bool
+        print rhs (val, units, label) in fixed-width cols?
+    varfmt: string
+        format for variable names
+    valfmt: string
+        format for scalar values
+    vecfmt: string
+        format for vector values
+    """
+    lines = []
+    decorated = [(bool(v.shape) if isinstance(v, Iterable) else False,
+                 ('%s : ' % k),
+                 i, k, v) for i, (k, v) in enumerate(data.items())
```

Comment:
This is the old "decorate-sort-undecorate" strategy, and the `i` makes the sort "stable", meaning that if two items happen to have the same vector-ness and str(key), their order will be preserved in the sorted list.

I'm not staying that's likely to happen here; rather I'm following a sorting recipe that guarantees stability.